add predefine DocumentPath placeholder

kdmcser opened the pull request:
In #5611 ,it's suggested to set `dirs.json` as follows:

```
{
    "userDataPath": "%USERPROFILE%\\Documents\\My Games\\vcmi",
    "userCachePath": "%USERPROFILE%\\Documents\\My Games\\vcmi\\cache",
    "userConfigPath": "%USERPROFILE%\\Documents\\My Games\\vcmi\\config",
    "userLogsPath": "%USERPROFILE%\\Documents\\My Games\\vcmi\\logs",
    "userSavePath": "%USERPROFILE%\\Documents\\My Games\\vcmi\\saves"
}
```

However, `%USERPROFILE%\\Documents\\` is not always the actual path to the user's My Documents folder. On Windows, users can change the location of My Documents, for example to `D:\My Document`.

Maybe we could introduce a predefined placeholder, such as `%MyDocuments%`, which VCMI can resolve at runtime to the correct path.

And function `SHGetSpecialFolderPathW` is Deprecated. Should replaced by `SHGetFolderPathW`.

kdmcser committed:
add predefine DocumentPath placeholder

GeorgeK1ng commented:
That's just example of dirs.json. VCMI itself don't use it until it's created. 

If you moved your directory, use hardcoded path. And if you did this move correctly, then it will be still loaded from CSIDL_PERSONAL without any change. It's connected with Shell Folders in registry.

And if is end user power user, then it's enough to add custom variable into PATH, then it will be correctly handled, no need to hardcode some kind of supported variables. That's why it have expand variables feature inside.

Btw mentioned function is also depricated and supported one is SHGetKnownFolderPath. But that doesn't matter at all. All 3 have same results.

I will try to add GUI that allows changing these folders into launcher soon.

kdmcser commented:
This feature is intended for mod pack integrators. They need to distribute their own mod packages to users and want those users to use a separate directory to avoid conflicts with other mod packs. However, they cannot assume the location of each user's "My Documents" folder—some users have changed it, while others have not. Therefore, they need VCMI to provide a unified way to retrieve the "My Documents" path, rather than relying on users to configure it themselves. Some users don't even know what an environment variable is. By using a built-in placeholder, their mod packages become much easier to use.

GeorgeK1ng commented:
What about OneDrive syncing? Keeping game data it in documents folder will still have same result in modern Windows. All files will goes online. That was one of case why was added support to change files location. 

Better is to use %AppData%, but then on server synced profile will payload there. 

So there will be always some kind of syncing issue. 

You can also use your portable VCMI directory, in dirs.json you can just add something like “vcmidata” into userDataPath and then will be used VCMI “application root\vcmidata”. Also if you don’t need to change other paths, you don’t need to specifiy them, userDataPath is enough.

kdmcser committed:
use CSIDL_PERSONAL instead of CSIDL_MYDOCUMENTS, and rename placeholder to USERDOCUMENTS

kdmcser commented:
In China, no one uses OneDrive. Additionally, maintaining document storage remains necessary. If users encounter issues, paths like %APPDATA% and %LOCALAPPDATA% are completely inaccessible to novice users - they can't cooperate when troubleshooting is required. If configured in the game installation directory, this data faces accidental deletion risks. Users habitually delete entire game directories when reinstalling new versions, leading to save file loss.

I've changed the placeholder names to %USERDOCUMENTS% and %MyDocument% as recommended by DeepSeek, but upon reflection, these don't seem appropriate either.

IvanSavenko commented:
> In China, no one uses OneDrive. 

Everyone else seems to use it, even though a lot of people are probably not aware of that since it is forced by MS nowadays.

> Additionally, maintaining document storage remains necessary. If users encounter issues, paths like %APPDATA% and %LOCALAPPDATA% are completely inaccessible to novice users

Those locations can be easily opened via Launcher

kdmcser commented:
```
	std::ifstream in(pathToUtf8(configPath), std::ios::binary);
	if (!in)
		return;

	std::string buffer((std::istreambuf_iterator<char>(in)), {});
	dirsConfig = std::make_unique<JsonNode>(reinterpret_cast<const std::byte*>(buffer.data()), buffer.size(), pathToUtf8(configPath));
```
This code will lead a problem: some people installed vcmi in A Chinese path. Their Windows use GBK, converting to UTF8 will make VCMI unable to find the real GBK path, and dirsConfig will remain empty.

kdmcser committed:
fix bug: setting windows encoding GBK is not support Chinese path in dirs

kdmcser commented:
This problem is fixed, but I found another encoding problem #5634,.

kambala-decapitator commented:
@kdmcser but what is the explanation regarding WideCharToMultibyte not working with GBK?

kambala-decapitator commented:
> When the resulting UTF-8 string is passed to std::ifstream (which expects ANSI-encoded paths on Windows)

Thanks, forgot about this

GeorgeK1ng commented:
This should be merged too to avoid issues in upcoming installer update with ability to select custom data folders locations. This PR should fix chinese characters in dirs.json.

mrhaandi committed:
fix rmg not respecting dwelling info and creating owned dwellings

mrhaandi committed:
take valueModifier for dwelling placement into account

mrhaandi committed:
exclude dwellings from ObjectDistributor::distributeLimitedObjects

mrhaandi committed:
adjust ownership in setBasicProperties

kdmcser committed:
add bonus FORCE_NEUTRAL_ENCOUNTER_STACK_COUNT

kdmcser committed:
fix compile

kdmcser committed:
fix json

kdmcser committed:
Fix crash when ATTACK_AND_RETURN unit is slowed by retaliation and can’t return to origin

kdmcser committed:
fix blind and bind after etaliate make it return bug

IvanSavenko committed:
Fix duration of bonuses from adventure map spells

IvanSavenko committed:
Fix possible crash on loading image with non-256 color palette

IvanSavenko committed:
Fix crash on importing mod from .zip

IvanSavenko committed:
Fix disassembling artifact when its part is needed for Seer Hut quest

IvanSavenko committed:
Respect 'special' flag when rolling for skills

kdmcser committed:
implement HotA map stack count

IvanSavenko committed:
Update build system for 1.7 release

- Github Actions: generate .aab with different build ID for x86_64
- Bump Android build ID since 1700 code was already uploaded
- Update dependencies for .deb packaging
- Disable MMAI for .deb packaging for now
- Switch Linux building docs to reference Qt6 packages for Ubuntu

kdmcser committed:
merge develop

Opuszek committed:
mapeditor - fixes crashes at picking victory conditions

IvanSavenko committed:
Merge pull request #6459 from IvanSavenko/release_prep

Update build system for 1.7 release

Laserlicht committed:
allow campaign videos without rim

Laserlicht committed:
checkbox for enable/disable rim in campaign editor

MichalZr6 committed:
Support system onnxruntime on Arch

IvanSavenko committed:
Merge pull request #6458 from IvanSavenko/beta_fixes

[1.7] Fixes for reported issues

Opuszek committed:
move object blit order comparator to library

Opuszek committed:
preserve instanceId during serialization/deserialization of json mapFormat

IvanSavenko committed:
Fix poor performance of RMG in recent builds

IvanSavenko committed:
Fix not hidden cursor on right-click

IvanSavenko committed:
Merge pull request #6468 from IvanSavenko/rmg_perf_fix

[1.7] Fix poor performance of RMG in recent builds

IvanSavenko committed:
Merge pull request #6467 from IvanSavenko/cursor_fix

[1.7] Fix not hidden cursor on right-click

IvanSavenko committed:
Merge pull request #6460 from Opuszek/fix_victory_condition_picker

mapeditor - fixes crashes at picking victory conditions

IvanSavenko committed:
Merge pull request #6466 from Opuszek/fix_editor_client_rendering_discrepancy

Fix editor client rendering discrepancy

IvanSavenko committed:
Merge pull request #6461 from Laserlicht/rim

[1.7.0] allow campaign videos without rim

Opuszek committed:
mapeditor fix changing neutral armies character

kdmcser committed:
support negative coord in mainmenu images and videos

IvanSavenko committed:
NKAI2 performance improvements

Performance improvements to functions that are called a lot by AI:
- simplify math in BuildAnalyzer
- remove complex randomization in NodeStorage
- Fix excessive copying in Creature::cost
- Generate list of all resources in ResourceTypeHandler once

IvanSavenko committed:
Fix COMBAT_MANA_BONUS when coming from town building

Unintended side effect from another PR. Added quick workaround to
account for COMBAT_MANA_BONUS coming from town or town buildings

IvanSavenko committed:
Merge pull request #6428 from kdmcser/neutral_enemy_stack_bonus

Add bonus FORCE_NEUTRAL_ENCOUNTER_STACK_COUNT for neutral stack splitting

IvanSavenko committed:
Merge pull request #6472 from Opuszek/fix_changing_neutral_armies_character

mapeditor fix changing neutral armies character

IvanSavenko committed:
Merge pull request #6462 from MichalZr6/onnx_for_arch

Support system onnxruntime on Arch

IvanSavenko committed:
Merge pull request #6331 from mrhaandi/fix-rmg-dwellings

Fix rmg not respecting dwelling info

IvanSavenko committed:
Fix loading of secondary skill specialties

Fixes bug where game would attempt to load specialty before completely
loading bonuses that are provided by skill that specialty affects

Now skill specialties are only loaded after identifier resolution stage
is complete

kdmcser committed:
Update client/mainmenu/CMainMenu.cpp

Co-authored-by: Ivan Savenko <saven.ivan@gmail.com>

kdmcser committed:
Update client/mainmenu/CMainMenu.cpp

Co-authored-by: Ivan Savenko <saven.ivan@gmail.com>

IvanSavenko committed:
Fix crash on loading single-level map with subterranean gates

IvanSavenko committed:
Merge pull request #6473 from kdmcser/mainmenu_negative_coord

Support negative coordinates for images and videos

IvanSavenko committed:
Merge pull request #6474 from IvanSavenko/combat_mana

Fix COMBAT_MANA_BONUS when coming from town building

IvanSavenko committed:
Merge pull request #6475 from IvanSavenko/ai_optimize

NKAI2 performance improvements

IvanSavenko committed:
Merge pull request #6479 from IvanSavenko/subterra_fix

Fix crash on loading single-level map with subterranean gates

IvanSavenko committed:
Merge pull request #6477 from IvanSavenko/hero_spec_fix

Fix loading of secondary skill specialties

kdmcser committed:
Merge branch 'develop' into attack_and_return_crash_fix

kdmcser committed:
try their best to return to start position

IvanSavenko committed:
Allow un-banning normally banned objects in RMG template

It is now possible to enabled heroes, spells, artifacts and skills that
are normally banned on a RMG template, using format similar to recently
added `bannedXXX : [ ... ]`

Also:
- made minor fixes to generate more clear error messages when identifier
resolution fails due to not specified mod ID for optional requests
- mod ID is now case-insensitive when requesting identifiers

ToRRent1812 committed:
🇵🇱 Translate everything for vcmi 1.7 + minor changes

ToRRent1812 committed:
Fix typo in damage reduction description

IvanSavenko committed:
Fix number of available skills if player has no empty skill slots

Laserlicht committed:
send logs functionality

Opuszek committed:
fixes to mapeditor

IvanSavenko committed:
Move University cost in gold to game setting

Quick change to remove hardcoded numbers. Value is shared between
Conflux and on-map universities, but can be changed in future

Laserlicht committed:
handle non loadable maps

Laserlicht committed:
add last save, skip persistentStorage.json, set home, rename button

Laserlicht committed:
fixes white line #6492

Laserlicht committed:
add vcmi version to gatherDeviceInfo; small improvements

IvanSavenko committed:
Fix crash on double-removal of artifact from defeated hero

May happen if:
- hero has artifact with single charge left
- this artifact is configured to be removed upon discharge
- this hero is defeated in battle against another hero

As result, game will attempt to both discharge & remove an artifact, as
well as transfer it to victor in combat

IvanSavenko committed:
Block attempt to select invalid custom size

- It is no longer possible to select invalid (too big or too small) map
size via custom size dialog when configuring random map
- Fixed bug where template name won't reset to default if template
parameters are no longer valid
- Custom map size dialog now displays supported map sizes for random map
template

Laserlicht committed:
github stars

Laserlicht committed:
sorting + repos & stars in text

Laserlicht committed:
fix mouse drag on higher scale factor

IvanSavenko committed:
Merge pull request #6481 from IvanSavenko/rmg_enable_objects

Allow un-banning normally banned objects in RMG template

IvanSavenko committed:
Merge pull request #6486 from IvanSavenko/secskill_fix

Fix number of available skills if player has no empty skill slots

IvanSavenko committed:
Merge pull request #6489 from Opuszek/mapeditor_fixes

fixes to mapeditor

IvanSavenko committed:
Merge pull request #6490 from IvanSavenko/university_config

[1.7.0] Move University cost in gold to game setting

IvanSavenko committed:
Merge pull request #6493 from Laserlicht/camp_conflict

[1.7.0] handle non loadable maps in campaign editor

IvanSavenko committed:
Merge pull request #6496 from IvanSavenko/rmg_custom_size_fix

Block attempt to select invalid custom size of random map

IvanSavenko committed:
Merge pull request #6497 from IvanSavenko/art_crash_fix

Fix crash on double-removal of artifact from defeated hero

IvanSavenko committed:
Merge pull request #6498 from Laserlicht/stars

[1.7.0] Show github stars in mod view

IvanSavenko committed:
Merge pull request #6491 from Laserlicht/logs

[1.7.0] export logs functionality

IvanSavenko committed:
Fixes and workarounds for crashes from Google Play

- Added more detailed error message for crash due to potential load of
mod with missing dependencies?
- Fix crash on shutting down server when playing through online lobby
- Log instead of crash when player (somehow) presses update button
without mod selected

IvanSavenko committed:
Auto-update VCMI Qt translation files

IvanSavenko committed:
Merge pull request #6504 from vcmi/bot/qt-translation-refresh/beta

Auto-update Qt translation files

IvanSavenko committed:
Update Ukrainian translation

IvanSavenko committed:
Merge pull request #6503 from IvanSavenko/crashfix

[1.7.0] Fixes and workarounds for crashes from Google Play

Laserlicht committed:
updated/optimized german

dydzio0614 committed:
Update vietnamese translation

Laserlicht committed:
fix resource

IvanSavenko committed:
Merge pull request #6508 from dydzio0614/viet-translation

Update vietnamese translation

IvanSavenko committed:
Merge pull request #6507 from Laserlicht/germ

[1.7.0] German update

IvanSavenko committed:
Merge pull request #6506 from IvanSavenko/translations

[1.7.0] Translations update

Laserlicht committed:
resize main window

Laserlicht committed:
inscreased window size

IvanSavenko committed:
Use markdown for mod descriptions and changelog

Replaces rich text / html usage with markdown formatting for mod
descriptions and changelog.

For mods that use html in descriptions, html will be converted into
markdown using Qt tools

In addition, mod description now show immediately below mod name,
instead of displaying it at the very end of mod information

Laserlicht committed:
fix sorting; fixed indention

Laserlicht committed:
reduce button size to avoid overlapping

IvanSavenko committed:
Merge pull request #6509 from Laserlicht/res_fix

[1.7.0] fix resource access

IvanSavenko committed:
Merge pull request #6494 from Laserlicht/fix_line

[1.7.0] fixes white line

IvanSavenko committed:
Merge pull request #6501 from Laserlicht/mouse_fix

[1.7.0] fix mouse drag on higher scale factor

IvanSavenko committed:
Merge pull request #6447 from kdmcser/attack_and_return_crash_fix

Fix crash when ATTACK_AND_RETURN unit is slowed by retaliation and can’t return to origin

Laserlicht committed:
Update launcher/modManager/modstateitemmodel_moc.cpp

Co-authored-by: Ivan Savenko <saven.ivan@gmail.com>

IvanSavenko committed:
Merge pull request #6511 from IvanSavenko/mod_markdown

[1.7.0?] Use markdown for mod descriptions and changelog

IvanSavenko committed:
Merge pull request #6510 from Laserlicht/res_main_win

[1.7.0] Increase standard window size

IvanSavenko committed:
Final changes for 1.7.0

- Updated changelog to include changes during beta
- Updated Polish translation as per comments in #6506
- Bumped Android build ID to next unused value
- Fixed missing property in game settings schema
- Workaround to remove 'scary' popups on launching game with 1.6 mods
- Replaced 1.6.X download counters with 1.7.0 counter

IvanSavenko committed:
Fix entering k/m/g/... symbols as 2nd character in number input

GeorgeK1ng committed:
Update installation help message for HD Edition

IvanSavenko committed:
Do not use non-terrain patch map objects as source for battlefield type

IvanSavenko committed:
Clarify creature level limiter docs

MichalZr6 committed:
Launcher and editor translation updates for 22 languages

GeorgeK1ng committed:
[1.7.0] Czech translation update (#6520)

* Update czech.json

* Update Czech translations for UI elements

Updated various translations in Czech language for UI elements and messages.

* Revise hints in english.json for clarity

Updated hints and messages for spell book and custom RMG size.

* Update czech.json

* Fix formatting in czech.json for quickSave help text

* Update english.json

* Update czech.json

* Update czech.json

MichalZr6 committed:
Merge branch 'beta' into translation_updates

IvanSavenko committed:
Apply review suggestions

IvanSavenko committed:
Fix crash on attempt to import preset with non-existing mod

IvanSavenko committed:
Merge pull request #6521 from MichalZr6/translation_updates

[Launcher/Map Editor]: update translations for 22 languages

IvanSavenko committed:
Merge pull request #6514 from IvanSavenko/release_170

[1.7.0] Final changes for 1.7.0

MichalZr6 committed:
Launcher and editor translation updates for 22 languages

IvanSavenko committed:
Merge pull request #6523 from vcmi/beta

Merge beta -> master

Opuszek committed:
mapeditor - allow gold as quest requirement and reward for a seer hut mission

GeorgeK1ng committed:
Update english.json

GeorgeK1ng committed:
Add translation for 'Ve stopách bohů' campaign

Laserlicht committed:
use star icon (CC0) in header

Laserlicht committed:
campaign set as CObjectListWindow

kambala-decapitator committed:
[iOS] support AI dylibs being frameworks

kambala-decapitator committed:
fix whitespace

smanolloff committed:
Ensure fallback triggers on any MMAI error; remove risky Render()

IvanSavenko committed:
Bump version to 1.7.1

IvanSavenko committed:
Add workaround for shifted bonus ID and assert to prevent this in future

IvanSavenko committed:
Fix loading of artifacts from .vmap's

IvanSavenko committed:
Fix crashes discovered on Google Play

IvanSavenko committed:
Merge pull request #6535 from Laserlicht/star

[1.7.1] use star icon instead of emoji

IvanSavenko committed:
Merge pull request #6532 from GeorgeK1ng/czech_update

[1.7.1] Missing strings for WoG campaign set

IvanSavenko committed:
Merge pull request #6524 from MichalZr6/translation_updates2

[Launcher/Map Editor] Translation updates for 22 languages - latest features

IvanSavenko committed:
Merge pull request #6525 from Opuszek/allow_gold_reward

[1.7.1] mapeditor - allow gold as quest requirement and reward for a seer hut

IvanSavenko committed:
Merge pull request #6518 from GeorgeK1ng/patch-2

[1.7.1] Update installation help message for HD Edition

IvanSavenko committed:
Merge pull request #6536 from Laserlicht/camp_set

[1.7.1] campaign set as CObjectListWindow

Laserlicht committed:
fixes some mods in combination with HD edition

IvanSavenko committed:
Slightly better fix for bonuses

smanolloff committed:
use dynamic input shapes

1. prevents errors on crowded battlefields (e.g. more than 20 stacks per hero)
2. improves compute efficiency (no more input padding)

To make use of the dynamic shapes, a new set of MMAI models is needed
(the new models will be released via an update through the mmai mod)

This code change is designed to be backward-compatible, so if we have:
- new VCMI + new mod: the dynamic and more efficient models will be used
- new VCMI + old mod: non-dynamic (less efficient) models will be used
- old VCMI + new mod: same as above
- old VCMI + old mod: same as above

Laserlicht committed:
bugfix for modlist

smanolloff committed:
address sonarcloud suggestions

IvanSavenko committed:
Merge pull request #6538 from kambala-decapitator/ios-dylib-frameworks

[iOS] support AI dylibs being frameworks

IvanSavenko committed:
Merge pull request #6554 from Laserlicht/bugfix

[1.7.1] fix modlist

IvanSavenko committed:
Merge pull request #6552 from Laserlicht/patch-1

[1.7.1] fixes some mods in combination with HD edition

IvanSavenko committed:
Merge pull request #6555 from smanolloff/mmai-dynshapes

MMAI performance & stability improvements

thehonestcto committed:
NK2: deactivate openMap and lower pathfinderBucketSize for faster AI turns until further performance improvements

IvanSavenko committed:
Fix crash on attempt to rollback stack rearrangement

IvanSavenko committed:
Fix rare case where mod without resolved depedencies added to load order

IvanSavenko committed:
Add workaround for crash on font rendering

IvanSavenko committed:
Add workaround for mod install crash

IvanSavenko committed:
Add potential fix for savegame list crash

IvanSavenko committed:
Fix crash on AI termination

IvanSavenko committed:
Fix crashing on drawing line outside of screen borders

IvanSavenko committed:
Workaround for inability to cast spells in 1.6 saves

IvanSavenko committed:
Fix two more potential crashes

IvanSavenko committed:
Merge pull request #6550 from IvanSavenko/hotfix

Fixes for critical 1.7.0 issues

OriginMD committed:
Update Installation_iOS.md

Updating scaling 273 -> 250

kambala-decapitator committed:
Merge pull request #6564 from OriginMD/patch-3

Update Installation_iOS.md

IvanSavenko committed:
Remove spaces in artifact file names for consistency

IvanSavenko committed:
Remove upper cap of damage since this logic breaks range randomization

Laserlicht committed:
mapeditor: fix passability & improve layout

IvanSavenko committed:
Fix deserialization of hero pool when mod load order changes

Laserlicht committed:
screenshot support

Laserlicht committed:
change to F3 (no dublicate)

IvanSavenko committed:
Merge pull request #6570 from Laserlicht/screenshot

[1.7.1] screenshot support

IvanSavenko committed:
Merge pull request #6568 from IvanSavenko/hotfix2

[1.7.1] Fixes for remaining critical 1.7.0 issues

IvanSavenko committed:
Merge pull request #6561 from thehonestcto/feature/nullkiller2-part2

NK2: deactivate openMap and lower pathfinderBucketSize for faster AI turns until further performance improvements

IvanSavenko committed:
Merge pull request #6569 from Laserlicht/maped

[1.7.1] mapeditor: fix passability & improve layout

Laserlicht committed:
replace replaceTextID with replaceRawString

Laserlicht committed:
template editor fixes

IvanSavenko committed:
Merge pull request #6572 from Laserlicht/replaceRawString

[1.7.1] replace replaceTextID with replaceRawString

IvanSavenko committed:
Refactor CSaveFile to prepare for future changes

IvanSavenko committed:
Potential fix for desync on AI turn

IvanSavenko committed:
Add option to disable before-visit save in Launcher

IvanSavenko committed:
Update changelog for 1.7.1

IvanSavenko committed:
Disable black bars on iOS by default

Laserlicht committed:
Apply suggestions from code review

Co-authored-by: Ivan Savenko <saven.ivan@gmail.com>

IvanSavenko committed:
Merge pull request #6582 from IvanSavenko/black_bars

[1.7.1] Disable black bars on iOS by default

IvanSavenko committed:
Merge pull request #6580 from IvanSavenko/before_visit_optional

[1.7.1?] Add option to disable before-visit save in Launcher

IvanSavenko committed:
Merge pull request #6577 from IvanSavenko/fix_hero_desync

Potential fix for desync on AI turn

IvanSavenko committed:
Merge pull request #6573 from Laserlicht/templ_editor

[1.7.1] template editor fixes

Karyoplasma committed:
refactor elixir of life limiter

+ fixes the bug that war machines were affected by the regeneration
+ shorter code

IvanSavenko committed:
Possible fix for crash on having growing and charging artifacts

IvanSavenko committed:
Fix assertion failure and invalid map state on prison release

IvanSavenko committed:
Reduce length of long text strings to fit into UI

IvanSavenko committed:
Fix reset of AI settings on Launcher start. Rename config keys to reset
settings for players

IvanSavenko committed:
Merge pull request #6581 from IvanSavenko/changelog_171

Update changelog for 1.7.1

IvanSavenko committed:
Bump android build ID to next unused value

IvanSavenko committed:
Final changelog update

IvanSavenko committed:
Merge pull request #6587 from Karyoplasma/fix-elixir-of-life

refactor elixir of life limiter

IvanSavenko committed:
Merge pull request #6589 from IvanSavenko/hotfix3

[1.7.1] Final changes for 1.7.1

IvanSavenko committed:
Fix settings schema format

IvanSavenko committed:
Merge pull request #6590 from IvanSavenko/fix_schema

Fix settings schema format

IvanSavenko committed:
Merge pull request #6591 from vcmi/beta

Merge beta -> master

kdmcser committed:
merge master

kdmcser committed:
fix compule
